src/scraper/scraper.cpp

```diff
@@ -304,6 +365,8 @@ void _log(logattribute eType, const std::string& sCall, const std::string& sMess
 
     log.output(sOut);
 
+    log.closelogfile();
```

Comment:
I disagree with closing the log file after each message. The file should be closed WHEN archiving it. Two threads writing to the same file with two handles and two logger objects is bad design. A file should be opened by one instance of logger only and that instance must have mutex to lock when archiving.

Reply:
If desired, the logging class can be split into backend and frontend. The backend holding only the mutex and file handle and frontend doing the formating and holding a reference to the backend. There can be then multiple frontends pointing to the same file and the archiving operation performed on the backend.

Reply:
What I don't like about that we are not just renaming it. We are also compressing the file for archiving. The compression could take a while. The advantage of the above approach is that the rename of the above file allows the archive process to work on the renamed file. The normal logging then immediately opens a new scraper.log file and continues writing and is not blocked by the compression process, which is acting on the renamed file.

Exerting a singleton pattern here will put the compression call in line with the rename, and the mutex will not be released for another use of the logger until the whole compression process is over.

So to recover that we would have to spawn a separate thread to do the compression. This is getting too complicated for the last minute to get Denise out the door.

Reply:
I obviously am not suggesting to complicate matters TAHT much. Only the close-rename-reopen needs to be under the lock of mutex.